### ISSUE TITLE
`reportMissingImports` raised in unreachable code behind platform check

### ISSUE THREAD

**Issue by Avasam** [microsoft/pyright#6156]
**Describe the bug**
`reportMissingImports` errors are raised despite being behind a platform check. It should be expected that some imports can be platform-specific.

**Code or Screenshots**
This snippet from https://github.com/python/typeshed/blob/main/tests/pytype_test.py . Run the following under Windows.

```python
import sys

if sys.platform == "win32":
    print("pytype does not support Windows.", file=sys.stderr)
    sys.exit(1)

from pytype import config as pytype_config, load_pytd
from pytype.imports import typeshed
```

![image](https://github.com/microsoft/pyright/assets/1350584/95d3d670-09a2-44fc-8855-4867a9408c04)

```
E:\Users\Avasam\Documents\Git\typeshed\tests\pytype_test.py
  E:\Users\Avasam\Documents\Git\typeshed\tests\pytype_test.py:37:6 - error: Import "pytype" could not be resolved (reportMissingImports)
  E:\Users\Avasam\Documents\Git\typeshed\tests\pytype_test.py:38:6 - error: Import "pytype.imports" could not be resolved (reportMissingImports)
```

**VS Code extension or command-line**
Are you running pyright as a VS Code extension, a language server in another editor, integrated into Pylance, or the command-line tool? Which version?

pyright@1.1.331

**Comment by erictraut** [comment]
This is by design, not a bug.

The determination about what code to analyze and what code to completely ignore is determined in the "binding" phase of the type checker long before types are evaluated and before the code flow graph is established. As per PEP 484, a type checker must honor a specific set of conditional checks including `sys.platform == <literal>`. You're correctly using this in your code, but you're not including the non-Windows code in an `else` statement. If you want this to work in pyright or other static type checking tools, you'd need to change your code as follows:

```python
import sys

if sys.platform == "win32":
    print("pytype does not support Windows.", file=sys.stderr)
    sys.exit(1)
else:
    from pytype import config as pytype_config, load_pytd
    from pytype.imports import typeshed
```
